framer/motion
# Fix per-axis viewport scroll blocking

Fixes https://github.com/framer/motion/issues/185
Fixes https://github.com/framer/motion/issues/310

By using e.preventDefault after a event has started, it's impossible to block the browser from touch-scrolling. So we do it regardless of the dragging axis.

Currently if you, for instance, set drag to `"x"`, it's impossible to start dragging vertically from that element. Not helpful for carousels etc. By using the info in the `drag` prop we can set a CSS `touch-action` rule that blocks the browser from triggering scroll in the other direction.

QA
- Run make dev
- Go to Drag-block-viewport-conditionally
- Turn on mobile view 
- Start dragging vertically from the white box - page should scroll
- Start dragging horizontally from white box - box should drag
